### ISSUE TITLE
Improve canonical data check script

### ISSUE THREAD

**Issue by FridaTveit** [exercism/java#1261]
The `canonical_data_check.sh` script is sometimes reporting that exercises need to have their version files updated when they actually don't (e.g. it outputs `"needs update (v1.0.0 -> v1.0.0)."`). I'm guessing this is because one or the other of the files ends with a new line or has an extra whitespace or something.

This should hopefully be relatively easy to fix by adding some trimming of whitespace or similar to the script.

**Comment by sjwarner-bp** [comment]
Ah, good spot!

If we do fix this up, it'll be worth offering the same changes to the Kotlin track, as I think they use a lot of the same scripts as us.

**Comment by Haozhe321** [comment]
@FridaTveit Any example of this? I ran a check, and couldn't find anything like that.

**Comment by FridaTveit** [comment]
Yes, when I ran it locally it reported that pretty much every exercise needed updating but for most of them the version hadn't actually changed.

**Comment by Haozhe321** [comment]
@FridaTveit Does yours look like mine? Terminal screenshot attached below.
I don't see any instances of `"needs update (v1.0.0 -> v1.0.0)."` where the current and updated version are the same.

![image](https://user-images.githubusercontent.com/26651556/35851718-cc7720c0-0b63-11e8-928a-23c1f559421f.png)

**Comment by sjwarner-bp** [comment]
It could be possible that problem-specifications have updated, so there are no longer instances of this.

I've not run this script for about a month, but when I get a moment I can try myself too :+1:

**Comment by FridaTveit** [comment]
@Haozhe321, no it did not look like that. Like @sjwarner-bp says, it's been a while since I ran it and changes have been made both to this repository and to problem-specifications since then. And it could be a problem that only occurs on windows. Either way, it did report exercises changed that hadn't been changed so it's worth looking into at some point. Though it's not very urgent since it's easy to tell from the output which exercises have actually changed 🙂

**Comment by Haozhe321** [comment]
@FridaTveit I see, thanks for the clarification :)
